test/lit.cfg

```diff
@@ -1370,36 +1369,21 @@ elif run_os == 'linux-androideabi' or run_os == 'linux-android':
     config.target_runtime = "native"
     config.target_swift_autolink_extract = inferSwiftBinary("swift-autolink-extract")
     config.target_sdk_name = "android"
-    android_link_paths_opt = "-L {} -L {} -L {}".format(
-        shell_quote(make_path(
-            config.android_ndk_path, "sources", "cxx-stl", "llvm-libc++",
-            "libs", ndk_platform_tuple)),
+    android_link_paths_opt = "-L {}".format(
         shell_quote(make_path(
             toolchain_directory, "lib", "gcc", ndk_platform_triple,
-            "{}.x".format(config.android_ndk_gcc_version))),
-        shell_quote(make_path(
-            toolchain_directory, ndk_platform_triple, "lib")))
-    # Since NDK r14 the headers are unified under $NDK_PATH/sysroot, so the -sdk
-    # switch is not enough. Additionally we have to include both the unified
-    # sysroot, and the architecture sysroot.
-    unified_android_include_path = shell_quote(make_path(
-        config.android_ndk_path, "sysroot", "usr", "include"))
-    architecture_android_include_path = shell_quote(make_path(
-        config.android_ndk_path, "sysroot", "usr", "include",
-        ndk_platform_triple))
-    android_include_paths_opt = "-I {} -I {}".format(
-        unified_android_include_path, architecture_android_include_path)
-    # clang can use -isystem, but Swift cannot.
-    android_include_system_paths_opt = "-isystem {} -isystem {}".format(
-        unified_android_include_path, architecture_android_include_path)
+            "{}.x".format(config.android_ndk_gcc_version))))
+    resource_dir_opt = ("-resource-dir %s" % test_resource_dir)
+    # Since NDK r19, the headers and libraries are available in a unified
+    # sysroot at $NDK_PATH/toolchains/llvm/prebuilt/$prebuilt_directory/sysroot,
+    # so the -sdk switch can now be used.
     config.target_build_swift = ' '.join([
         config.swiftc,
         '-target', config.variant_triple,
-        '-Xcc', '--sysroot={}'.format(config.variant_sdk),
-        '-Xclang-linker', '--sysroot={}'.format(config.variant_sdk),
+        '-sdk', config.variant_sdk, '-Xclang-linker',
+        '--target={}{}'.format(config.variant_triple, config.android_api_level),
```

Comment:
This fixes the issues you saw with finding the right C libraries, without having to set an Android API level in the Swift target triple.